File: Observer/CPlusPlus/ObserverPattern/Subscriber.cpp
#include "Subscriber.h"

Subscriber::Subscriber(std::string name)
{
	this->name = name;
}
Subscriber::~Subscriber()
{

}
void Subscriber::update(std::string message)
{
	this->message = message;
	std::cout << "this is a message from a subscriber which is called " << name
		<< ". message is " << message << std::endl;
}

File: Observer/Java/ObserverPattern/src/observerpattern/Subscriber.java

package observerpattern;

public class Subscriber implements INewsFlash
{
    private String name;
    private String news;
    
    public Subscriber(String name)
    {
        this.name = name;
    }

    @Override
    public void update(String newsFlash) {
        this.news = newsFlash;
        getNews();
    }
    
    private void getNews()
    {
        System.out.println("this subscriber called " + name + " has just got this news flash " + news);
    }
}
